internal/weaver/stub.go

```diff
@@ -43,15 +44,20 @@ func (s *stub) Tracer() trace.Tracer {
 }
 
 // Run implements the codegen.Stub interface.
-func (s *stub) Run(ctx context.Context, method int, args []byte, shardKey uint64) ([]byte, error) {
+func (s *stub) Run(ctx context.Context, method int, args []byte, shardKey uint64) (result []byte, err error) {
 	m := s.methods[method]
 	opts := call.CallOptions{
-		// TODO(sanjay): Enable retries after weavertest supports retry injection
-		// so we can spot method methods that should be marked as non-retriable.
-		// Retry:    m.retry,
+		Retry:    m.retry,
 		ShardKey: shardKey,
 	}
-	return s.conn.Call(ctx, m.key, args, opts)
+	n := 1
+	if m.retry {
+		n += s.injectRetries
+	}
+	for i := 0; i < n; i++ {
+		result, err = s.conn.Call(ctx, m.key, args, opts)
```

Comment:
Q: Shouldn't retry logic have some sort of backoff e.g. exponential?

Reply:
Hi @pmatseykanets! Good question. The real retry logic does have exponential backoff and is performed within a call to `s.conn.Call`: 

https://github.com/ServiceWeaver/weaver/blob/2ed7c7f552b4b90b802fac25518a138325571e82/internal/net/call/call.go#L377-L384

This code is artificially injecting fake retries into tests, which doesn't need backoff.

Reply:
@mwhittaker Nice! Thank you so much for explaining!

Reply:
Thanks for the comment. Probably worth explaining with a comment, which I will do.